edx_proctoring/static/proctoring/js/exam_action_handler.js

```diff
@@ -0,0 +1,95 @@
+var edx = edx || {};
+
+(function($) {
+  'use strict';
+
+  var actionToMessageTypesMap = {
+    'submit': {
+      promptEventName: 'endExamAttempt',
+      responseEventName: 'examAttemptEnded'
+    },
+    'start': {
+      promptEventName: 'startExamAttempt',
+      responseEventName: 'examAttemptStarted'
+    }
+  };
+
+  function workerPromiseForEventNames(eventNames) {
+    return function() {
+      var proctoringBackendWorker = new Worker(edx.courseware.proctored_exam.configuredWorkerURL);
+      return new Promise(function(resolve) {
+        var responseHandler = function(e) {
+          if (e.data.type === eventNames.responseEventName) {
+            proctoringBackendWorker.removeEventListener('message', responseHandler);
+            proctoringBackendWorker.terminate();
+            resolve();
+          }
+        };
+        proctoringBackendWorker.addEventListener('message', responseHandler);
```

Comment:
Why not use the event name here for the eventListener rather than the generic 'message' - it may make debugging easier in the future and would allow active listeners for multiple event types to be created in the future.

Reply:
IIUC this is standard webworker boilerplate; `MessageEvent`s are all that are propagated between a page and the web workers they spawn. I don't think I can listen to any other types of events here.